api/include/opentelemetry/common/attribute_value.h

```diff
@@ -10,46 +10,67 @@
 OPENTELEMETRY_BEGIN_NAMESPACE
 namespace common
 {
+/// OpenTelemetry signals can be enriched by adding attributes. The
+/// \c AttributeValue type is defined as a variant of all attribute value
+/// types the OpenTelemetry C++ API supports.
+///
+/// The following attribute value types are supported by the OpenTelemetry
+/// specification:
+///  - Primitive types: string, boolean, double precision floating point
+///    (IEEE 754-1985) or signed 64 bit integer.
+///  - Homogenous arrays of primitive type values.
+///
+/// \warning
+/// \parblock The OpenTelemetry C++ API currently supports several attribute
+/// value types that are not covered by the OpenTelemetry specification:
+///  - \c uint64_t
+///  - \c nostd::span<const uint64_t>
+///  - \c nostd::span<uint8_t>
+///
+/// Those types are reserved for future use and currently should not be
+/// used. There are no guarantees around how those values are handled by
+/// exporters.
+/// \endparblock
 using AttributeValue =
     nostd::variant<bool,
                    int32_t,
                    int64_t,
                    uint32_t,
-                   uint64_t,
                    double,
                    nostd::string_view,
-#ifdef HAVE_SPAN_BYTE
-                   // TODO: 8-bit byte arrays / binary blobs are not part of OT spec yet!
-                   // Ref: https://github.com/open-telemetry/opentelemetry-specification/issues/780
-                   nostd::span<const uint8_t>,
-#endif
                    nostd::span<const bool>,
                    nostd::span<const int32_t>,
                    nostd::span<const int64_t>,
                    nostd::span<const uint32_t>,
-                   nostd::span<const uint64_t>,
                    nostd::span<const double>,
-                   nostd::span<const nostd::string_view>>;
+                   nostd::span<const nostd::string_view>,
+                   // Not currently supported by the specification, but reserved for future use.
+                   // Added to provide support for all primitive C++ types.
+                   uint64_t,
+                   // Not currently supported by the specification, but reserved for future use.
+                   // Added to provide support for all primitive C++ types.
+                   nostd::span<const uint64_t>,
+                   // Not currently supported by the specification, but reserved for future use.
```

Comment:
nit: this comment is duplicated with the above one at line 47?

Reply:
Just added it to every variant type that is not supported, to avoid any misunderstandings.